test/Npgsql.Tests/Types/TextTests.cs

```diff
@@ -229,26 +229,35 @@ public async Task Citext()
         [Test]
         public async Task Xml()
         {
-            using var conn = await OpenConnectionAsync();
-            using var cmd = new NpgsqlCommand("SELECT @p1, @p2", conn);
-            const string expected = "<root>foo</root>";
-            var p1 = new NpgsqlParameter("p1", NpgsqlDbType.Xml);
-            var p2 = new NpgsqlParameter("p2", DbType.Xml);
-            Assert.That(p1.NpgsqlDbType, Is.EqualTo(NpgsqlDbType.Xml));
-            Assert.That(p2.DbType, Is.EqualTo(DbType.Xml));
-            cmd.Parameters.Add(p1);
-            cmd.Parameters.Add(p2);
-            p1.Value = p2.Value = expected;
-            using var reader = await cmd.ExecuteReaderAsync();
-            reader.Read();
+            try
+            {
+                using var conn = await OpenConnectionAsync();
+                using var cmd = new NpgsqlCommand("SELECT @p1, @p2", conn);
+                const string expected = "<root>foo</root>";
+                var p1 = new NpgsqlParameter("p1", NpgsqlDbType.Xml);
+                var p2 = new NpgsqlParameter("p2", DbType.Xml);
+                Assert.That(p1.NpgsqlDbType, Is.EqualTo(NpgsqlDbType.Xml));
+                Assert.That(p2.DbType, Is.EqualTo(DbType.Xml));
+                cmd.Parameters.Add(p1);
+                cmd.Parameters.Add(p2);
+                p1.Value = p2.Value = expected;
+                using var reader = await cmd.ExecuteReaderAsync();
+                reader.Read();
 
-            for (var i = 0; i < cmd.Parameters.Count; i++)
+                for (var i = 0; i < cmd.Parameters.Count; i++)
+                {
+                    Assert.That(reader.GetFieldType(i), Is.EqualTo(typeof(string)));
+                    Assert.That(reader.GetDataTypeName(i), Is.EqualTo("xml"));
+                    Assert.That(reader.GetString(i), Is.EqualTo(expected));
+                    Assert.That(reader.GetFieldValue<string>(i), Is.EqualTo(expected));
+                    Assert.That(reader.GetValue(i), Is.EqualTo(expected));
+                }
+            }
+            catch (PostgresException e) when (e.SqlState == "0A000"/*feature_not_supported*/)
             {
-                Assert.That(reader.GetFieldType(i), Is.EqualTo(typeof(string)));
-                Assert.That(reader.GetDataTypeName(i), Is.EqualTo("xml"));
-                Assert.That(reader.GetString(i), Is.EqualTo(expected));
-                Assert.That(reader.GetFieldValue<string>(i), Is.EqualTo(expected));
-                Assert.That(reader.GetValue(i), Is.EqualTo(expected));
+                throw new IgnoreException(
+                    "The XML data type is not supported by your database." +
```

Comment:
What PG comes without the XML type??

But OK... It still seems better to do an upfront check here, rather than to catch the exception (which seems vague enough to possibly correspond to something else. For example, you could check `conn.TypeMapper.Mappings.Any(m => m.PgTypeName == "xml")`.

Reply:
> What PG comes without the XML type??

Mine 😁.
If you compile PostgreSQL yourself libxml2 is just another dependency you have to pull in before building.
If you don't do that, xml isn't supported.

While this is indeed a rare condition I think that it doesn't hurt us to skip the test if it isn't supported because this obviously is not what we are testing for and a test failure in that case doesn't indicate a problem within Npgsql.


> For example, you could check `conn.TypeMapper.Mappings.Any(m => m.PgTypeName == "xml")`.

👍🏻

Reply:
Makes sense :)

Reply:
> 👍🏻

It turns out that isn't that easy:

1. Multiplexing: `System.NotSupportedException : Connection-specific type mapping is unsupported when multiplexing is enabled.`
2. Even in non-multiplexing `pg_type` contains all information about the xml type even if it isn't supported.

I don't see any way but catching the exception but we can do so upfront with a very limited statement to reduce the probability that it covers up an actual bug.